main/src/libraries/wrdac/include/wrdac/subsystems/subSystem_ARE.h

```diff
@@ -49,21 +50,70 @@ namespace wysiwyd {
 
             SubSystem_Attention* SubATT;
             bool ATTconnected;
-        
+
             yarp::os::RpcClient cmdPort;
             yarp::os::RpcClient rpcPort;
             yarp::os::RpcClient getPort;
+            yarp::os::RpcClient d2kPort;
+
+            std::string lastlyUsedHand;
 
             /********************************************************************************/
             void appendCartesianTarget(yarp::os::Bottle& b, const yarp::sig::Vector &t)
             {
-                yarp::os::Bottle &sub = b.addList();
+                yarp::os::Bottle &sub=b.addList();
                 sub.addString("cartesian");
-                for (size_t i = 0; i < t.length(); i++)
+                for (size_t i=0; i<t.length(); i++)
                     sub.addDouble(t[i]);
             }
 
             /********************************************************************************/
+            void selectHandCorrectTarget(yarp::os::Bottle& options, yarp::sig::Vector& target,
+                                         const std::string handToUse="")
+            {
+                std::string hand="";
+                for (int i=0; i<options.size(); i++)
+                {
+                    yarp::os::Value val=options.get(i);
+                    if (val.isString())
+                    {
+                        std::string item=val.asString();
+                        if ((item=="left") || (item=="right"))
+                            hand=item;
+                    }
+                }
+
+                // always choose hand
+                if (hand.empty())
+                {
+                    if (handToUse.empty())
+                    {
+                        hand=(target[1]>0.0?"right":"left"); 
+                        options.addString(hand.c_str());
+                    }
+                    else
+                        hand=lastlyUsedHand;
```

Comment:
Actually, `lastlyUsedHand` serves a different purpose: to know which was the hand used last time for checking the correct hand in `dropOn`.